### ISSUE TITLE
Allow to set file filename during upload

### ISSUE THREAD

**Issue by markvaneijk** [uploadcare/uploadcare-php#115]
I can't seem to find how to set the filename for the uploaded file using this library? Using the API it's possible, but not using this package it seems..

**Comment by dmitry-mukhin** [comment]
`Uploadcare\File()->data` contains the properties fetched from API
You should be able to do the following:

```php
$file = $api->getFile($uuid);
$fn = $file->data['original_filename'];
```

**Comment by markvaneijk** [comment]
I actually mean when _uploading_ via the uploader class using the 'fromContents' or 'fromPath' methods, I can't provide the original filename.

So not fetching a file already uploaded to Uploadcare but uploading a file with an filename to Uploadcare.

**Comment by dmitry-mukhin** [comment]
gotcha.
but what's up with issue title?

**Comment by dmitry-mukhin** [comment]
also, can you please elaborate on a use case when you need to override a filename?

**Comment by markvaneijk** [comment]
Weird thing about the title, but now it's better :)

The thing is this..

When I upload using the widget on my Uploadcare account I get this as a result:

<img width="749" alt="Screenshot 2019-08-22 at 17 16 19" src="https://user-images.githubusercontent.com/1925388/63527139-d4a89280-c500-11e9-93e3-39aed4653298.png">

When I upload via the PHP package (where I can't set the filename as far as I can see), I get this:

<img width="737" alt="Screenshot 2019-08-22 at 17 17 15" src="https://user-images.githubusercontent.com/1925388/63527177-e1c58180-c500-11e9-8969-443fc44786b7.png">

So the file does not have an extension and I can't seem to provide an alternative filename. This way I also can't download/view the file as an PDF using the CDN link from Uploadcare.

**Comment by dmitry-mukhin** [comment]
FYI, you can use anything as a filename, when using CDN. Just add what you want after the last slash:
https://uploadcare.com/docs/delivery/file_api/#cdn-filename

**Comment by markvaneijk** [comment]
Already tried that..

For example this file: https://ucarecdn.com/b73e4b60-a961-4275-b118-e40e76565480/
Custom filename in URL: https://ucarecdn.com/b73e4b60-a961-4275-b118-e40e76565480/example.pdf

The above does not work in Google Chrome:

<img width="1131" alt="Screenshot 2019-08-22 at 17 24 59" src="https://user-images.githubusercontent.com/1925388/63527682-d58df400-c501-11e9-8a1e-5ccc75f1efc5.png">

It works better in Safari, where I can at least download the file and it recognises that it's a PDF. But the download filename is still the 'ucrHEGHoQ.pdf' instead of 'example.pdf'

**Comment by dmitry-mukhin** [comment]
it obviously works :) Not sure what content you're having in that doc that makes chrome show the warning. Or some settings in Chrome itself.

```
curl -I https://ucarecdn.com/b73e4b60-a961-4275-b118-e40e76565480/example.pdf
HTTP/1.1 200 OK
Accept-Ranges: bytes
Access-Control-Allow-Methods: HEAD, GET, OPTIONS
Access-Control-Allow-Origin: *
Content-Disposition: attachment; filename=ucrHEGHoQ
Content-Type: application/pdf
ETag: "c5c8936d98e5083c1ab27ea9bf2e77b4"
Last-Modified: Thu, 22 Aug 2019 15:17:15 GMT
Server: nginx
Content-Length: 4345
Cache-Control: public, max-age=31556513
Date: Thu, 22 Aug 2019 16:08:27 GMT
Connection: keep-alive
```

**Comment by markvaneijk** [comment]
Obviously my users/visitors use Chrome and not the cli with curl ;)

It happens with all kinds and different PDF files.

Also: I did not change any settings in Chrome. Please just try it in the browser. I can see where the problem is, it is in this part which communicates the filename on the project and not that is appended to the URL:

‘’’Content-Disposition: attachment; filename=ucrHEGHoQ’’’

Browsers look at headers and not in the filename in the URL.

**Comment by dmitry-mukhin** [comment]
The fact that header is not taking "filename" part into account should be fixed on backend side.
Meanwhile, feel free to send a PR that adds "filename" argument to uploading method.

**Comment by markvaneijk** [comment]
I will create a PR for the addition of the filename when uploading.

When do you think the backend is updated? Currently I see the same situation here:

<img width="621" alt="Screenshot 2019-08-23 at 13 52 50" src="https://user-images.githubusercontent.com/1925388/63590852-6a99f700-c5ad-11e9-98aa-104fe00d4595.png">

**Comment by dmitry-mukhin** [comment]
I can't say when it will be addressed at all, unfortunately.

**Comment by markvaneijk** [comment]
Oh ok, I thought you meant that you've changed something on the backend when you said:

"The fact that header is not taking "filename" part into account should be fixed on backend side."

Or do you mean you don't know when it will be live?

**Comment by dmitry-mukhin** [comment]
This is something that has to be done in CDN's origin code.
I can't say when it will be done.

**Comment by markvaneijk** [comment]
I think this solves this issue #116

**Comment by markvaneijk** [comment]
> This is something that has to be done in CDN's origin code.
> I can't say when it will be done.

Ok, with my pull request the biggest problem on this is solved by passing in a correct filename on upload using the API. So I can make sure on before hand that the filename and extension is correct.

It would be nice though to be able to dynamically change the filename via the URL, which was suggested as a feature, but this is a little bit broken then for now.

**Comment by dmitry-mukhin** [comment]
Addressed in #116

**Comment by dmitry-mukhin** [comment]
Closed in #116
